src/datasets/dataset_dict.py

```diff
@@ -478,31 +481,51 @@ def shuffle(
             }
         )
 
-    def save_to_disk(self, dataset_dict_path: str):
+    def save_to_disk(self, dataset_dict_path: str, fs=None):
         """
-        Save the dataset dict in a dataset dict directory.
+        Saves a dataset dict to a filesystem using either :class:`datasets.filesystem.S3FileSystem` or ``fsspec.spec.AbstractFileSystem``.
 
         Args:
-            dataset_dict_path (``str``): path of the dataset dict directory where the dataset dict will be saved to
+            dataset_dict_path (``str``): path (e.g. ``dataset/train``) or remote uri (e.g. ``s3://my-bucket/dataset/train``) of the dataset dict directory where the dataset dict will be saved to
+            fs (Optional[:class:`datasets.filesystem.S3FileSystem`,``fsspec.spec.AbstractFileSystem``],  `optional`, defaults ``None``): instance of :class:`datasets.filesystem.S3FileSystem` or ``fsspec.spec.AbstractFileSystem`` used to download the files from remote filesystem.
         """
-        os.makedirs(dataset_dict_path, exist_ok=True)
+        if is_remote_filesystem(fs):
+            proc_dataset_dict_path = preproc_dataset_path(dataset_dict_path)
+        else:
+            fs = fsspec.filesystem("file")
+            proc_dataset_dict_path = dataset_dict_path
+
+        os.makedirs(proc_dataset_dict_path, exist_ok=True)
```

Comment:
We only need to create it is the destination is a local path.
For remote path we should use a temp dir for the upload as in my previous comment.